modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceOrderManageApi.java

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param serviceId id of the service.
+     */
+    @Tag(name = "ServiceOrders",
+            description = "APIs to manage orders of services")
+    @Operation(description = "Delete all service orders of the service.")
+    @DeleteMapping(value = "/services/{serviceId}/orders",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @AuditApiRequest(methodName = "getCspFromServiceId")
+    public void deleteOrdersByServiceId(
+            @Parameter(name = "serviceId", description = "Id of the service")
+            @PathVariable(name = "serviceId") String serviceId) {
+        serviceOrderManager.deleteOrdersByServiceId(UUID.fromString(serviceId));
+    }
+
+
+    /**
+     * Get details of the service order by the order id.
+     *
+     * @param orderId id of the service order.
+     * @return order details.
+     */
+    @Tag(name = "ServiceOrders",
+            description = "APIs to manage orders of services")
+    @Operation(description = "Get details of the service order by the order id.")
+    @GetMapping(value = "/services/orders/{orderId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @AuditApiRequest(methodName = "getCspFromServiceOrderId")
+    public ServiceOrderDetails getOrderDetailsByOrderId(
+            @Parameter(name = "orderId", description = "Id of the service order")
+            @PathVariable(name = "orderId") String orderId) {
+        return serviceOrderManager.getOrderDetailsByOrderId(UUID.fromString(orderId));
+    }
+
+    /**
+     * Delete service order by the order id.
+     *
+     * @param orderId id of the service order.
+     */
+    @Tag(name = "ServiceOrders",
+            description = "APIs to manage orders of services")
+    @Operation(description = "Delete the service order by the order id.")
+    @DeleteMapping(value = "/services/orders/{orderId}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @AuditApiRequest(methodName = "getCspFromServiceOrderId")
+    public void deleteOrderByOrderId(
+            @Parameter(name = "orderId", description = "Id of the service order")
+            @PathVariable(name = "orderId") String orderId) {
+        serviceOrderManager.deleteOrderByOrderId(UUID.fromString(orderId));
+    }
+
+
+    /**
+     * Method to fetch status of service deployment.
+     */
+    @Tag(name = "ServiceOrders",
+            description = "APIs to manage orders of services")
+    @GetMapping(value = "/services/orders/{orderId}/status",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(description =
+            "Long-polling method to get the latest or updated task status of the service order.")
+    @AuditApiRequest(methodName = "getCspFromServiceOrderId")
+    @ResponseStatus(HttpStatus.OK)
+    public DeferredResult<ServiceOrderStatusUpdate> getLatestServiceOrderStatus(
```

Comment:
This same method is available in ServiceOrderManageApi also?

Reply:
This new one contains error message is for getting the  task status by the order id. The old one just moved into `ServiceDeployApi` is for getting the  deployment state by the service id.